src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java

```diff
@@ -281,6 +284,36 @@ public static CollideEntityEvent callCollideEntityEvent(net.minecraft.world.Worl
         return event;
     }
 
+    /**
+     * This simulates the blocks a piston moves and calls the event for saner debugging.
+     * @return if the event was cancelled
+     */
+    public static boolean handlePistonEvent(IMixinWorldServer world, WorldServer.ServerBlockEventList list, Object obj, BlockPos pos, Block blockIn, int eventId, int eventParam) {
+        boolean extending = (eventId == 0);
+        final IBlockState blockstate = ((net.minecraft.world.World)world).getBlockState(pos);
+        EnumFacing direction = (EnumFacing) blockstate.getValue(BlockDirectional.FACING);
+
+        List<BlockPos> blocks = new ArrayList<BlockPos>();
+        blocks.add(pos);
+
+        BlockPistonStructureHelper movedBlocks = new BlockPistonStructureHelper((WorldServer) (Object) world, pos, direction, extending);
+        movedBlocks.canMove(); // calculates blocks to be moved
+        blocks.addAll(movedBlocks.getBlocksToDestroy());
+        blocks.addAll(movedBlocks.getBlocksToMove());
+
+        List<Location<org.spongepowered.api.world.World>> locations = new ArrayList<Location<org.spongepowered.api.world.World>>();
```

Comment:
You're creating unnecessary lists and creating unnecessary objects, yes it may look a little cluttered, but it minimizes the amount of objects created.